app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -8,32 +8,62 @@ def create_in_provider(manager_id, params)
         connection.api.credentials.create!(params)
       end
 
+      refresh(manager)
+      find_by!(:resource_id => manager.id, :manager_ref => credential.id)
+    end
+
+    def create_in_provider_queue(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      queue(manager.my_zone, "create_in_provider", [manager_id, params], "Creating #{name}")
+    end
+
+    private
```

Comment:
You'll have to do `private_class_method :method_name` in order to get private class methods.